test/unit/eval_algorithms/test_factual_knowledge.py

```diff
@@ -40,6 +44,20 @@ def test_factual_knowledge_invalid_config(self):
         with pytest.raises(EvalAlgorithmClientError, match=re.escape(expected_error_message)):
             FactualKnowledgeConfig(target_output_delimiter="")
 
+    def test_factual_knowledge_invalid_config_with_and(self):
+        """
+        GIVEN invalid inputs
+        WHEN FactualKnowledgeConfig is initialized
+        THEN correct exception with proper message is raised
+        """
+        logical_operator_error_message = (
+            'Invalid logical_operator is provided. The only valid inputs are strings "OR" and "AND".'
+        )
+        invalid_inputs = ["", "NOT AND", "Random", "<OR>"]
+        for invalid_input in invalid_inputs:
+            with pytest.raises(EvalAlgorithmClientError, match=re.escape(logical_operator_error_message)):
+                FactualKnowledgeConfig(logical_operator=invalid_input)
+
     class TestCaseFactualKnowledgeEvaluateSample(NamedTuple):
```

Comment:
To avoid code duplication, I'd suggest to add one more field `logic_operator` to `TestCaseFactualKnowledgeEvaluateSample`, for existing test it can be "OR", and for the cases that you added it will be "AND".